src/testnet/helium/event_dispatcher.rs

```diff
@@ -0,0 +1,242 @@
+        let sock_addr = SocketAddr::new(address.parse().unwrap(), port);
+        EventObserver { sock_addr }
+    }
+
+    pub fn send(&mut self, filtered_events: Vec<&(Txid, &StacksTransactionEvent)>, chain_tip: &StacksBlock, chain_tip_info: &StacksHeaderInfo, receipts: &Vec<StacksTransactionReceipt>) {
+        // Initiate a tcp socket
+        let stream = TcpStream::connect(&self.sock_addr).unwrap();
+        
+        // Serialize events to JSON
+        let serialized_events: Vec<serde_json::Value> = filtered_events.iter().map(|(txid, event)| 
+            match event {
+                StacksTransactionEvent::SmartContractEvent(event_data) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "contract_event",
+                    "contract_event": {
+                        "contract_identifier": event_data.key.0.to_string(),
+                        "topic": event_data.key.1,
+                        "value": event_data.value,
+                    }
+                }),
+                StacksTransactionEvent::STXEvent(STXEventType::STXTransferEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "stx_transfer_event",
+                    "stx_transfer_event": event_data
+                }),
+                StacksTransactionEvent::STXEvent(STXEventType::STXMintEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "stx_mint_event",
+                    "stx_mint_event": event_data
+                }),
+                StacksTransactionEvent::STXEvent(STXEventType::STXBurnEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "stx_burn_event",
+                    "stx_burn_event": event_data
+                }),
+                StacksTransactionEvent::NFTEvent(NFTEventType::NFTTransferEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "nft_transfer_event",
+                    "nft_transfer_event": event_data
+                }),
+                StacksTransactionEvent::NFTEvent(NFTEventType::NFTMintEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "nft_mint_event",
+                    "nft_mint_event": event_data
+                }),
+                StacksTransactionEvent::FTEvent(FTEventType::FTTransferEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "ft_transfer_event",
+                    "ft_transfer_event": event_data
+                }),
+                StacksTransactionEvent::FTEvent(FTEventType::FTMintEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "ft_mint_event",
+                    "ft_mint_event": event_data
```

Comment:
The `amount: u128` fields in these `event_data` structs are unsafely serialized into JSON numbers. 

For example, using this value in rust/clarity `amount: 340282366920938463463374607431768211455` gets converted into the js number `3.402823669209385e+38` after `JSON.parse`.

These should be a hex string or a quoted number string.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1319/commits/2f3e8c8f8389aa02d99e1257f1a57ee2e49b24bb